wrapanapi/rhevm.py

```diff
@@ -562,3 +533,190 @@ def usage_and_quota(self):
+            if image.name == image_name:
+                return self._get_images_service(storage_domain_name).image_service(image.id)
+
+    def import_glance_image(self, storage_domain_name, cluster_name, temp_template_name,
+                            template_name, async=True, import_as_template=True):
+        image_service = self._get_image_service(storage_domain_name, template_name)
+        image_service.import_(
+            async=async,
+            import_as_template=import_as_template,
+            template=types.Template(name=temp_template_name),
+            cluster=types.Cluster(name=cluster_name),
+            storage_domain=types.StorageDomain(name=storage_domain_name)
+        )
+        wait_for(self.does_template_exist, func_args=[temp_template_name], delay=5, num_sec=240)
+
+    def _get_disk_attachments_service(self, vm_name):
+        vm_id = self.get_vm_guid(vm_name)
+        return self._vms_service.vm_service(vm_id).disk_attachments_service()
+
+    def _get_disk_attachment_service(self, vm_name, disk_name):
+        disk_attachments_service = self._get_disk_attachments_service(vm_name)
+        for disk_attachment_service in disk_attachments_service.list():
+            disk = self.api.follow_link(disk_attachment_service.disk)
+            if disk.name == disk_name:
+                return disk_attachments_service.service(disk.id)
+        raise ItemNotFound(disk_name, 'disk')
+
+    def is_disk_attached_to_vm(self, vm_name, disk_name):
+        try:
+            return bool(self._get_disk_attachment_service(vm_name, disk_name))
+        except ItemNotFound:
+            return False
+
+    def get_vm_disks_count(self, vm_name):
+        return len(self._get_disk_attachments_service(vm_name).list())
+
+    def _get_disk_service(self, disk_name):
+        disks_service = self.api.system_service().disks_service()
+        query_result = disks_service.list(search="name={}".format(disk_name))
+        if not query_result:
+            raise ItemNotFound(disk_name, 'disk')
+        else:
+            disk = query_result[0]
+            return disks_service.service(disk.id)
+
+    def does_disk_exist(self, disk_name):
+        try:
+            return bool(self._get_disk_service(disk_name))
+        except ItemNotFound:
+            return False
+
+    def _check_disk(self, disk_id):
+        disks_service = self.api.system_service().disks_service()
+        disk_service = disks_service.disk_service(disk_id)
+        disk = disk_service.get()
+        return disk.status == types.DiskStatus.OK
+
+    def add_disk_to_vm(self, vm_name, storage_domain=None, size=None, interface=None, format=None,
+            active=True):
+        disk_attachments_service = self._get_disk_attachments_service(vm_name)
+        disk_attachment = disk_attachments_service.add(
+            types.DiskAttachment(
+                disk=types.Disk(
+                    format=format,
```

Comment:
This needs to be a `DiskFormat` type object, which I don't think we can expect callers to provide. Raw strings passed as format result in:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-16-7e212df1967d> in <module>()
----> 1 rhv.add_disk_to_vm('dmisharo_delete_renamed', storage_domain='env-rhv41-01-nfs-data', size=1, format='raw')

/home/mshriver/repos/wrapanapi/wrapanapi/rhevm.py in add_disk_to_vm(self, vm_name, storage_domain, size, interface, format, active)
    664                     storage_domains=[
    665                         types.StorageDomain(
--> 666                             name=storage_domain,
    667                         )
    668                     ]

/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/ovirtsdk4/types.pyc in __init__(self, active, actual_size, alias, bootable, comment, content_type, description, disk_profile, format, id, image_id, initial_size, instance_type, interface, logical_name, lun_storage, name, openstack_volume_type, permissions, propagate_errors, provisioned_size, qcow_version, quota, read_only, sgio, shareable, snapshot, sparse, statistics, status, storage_domain, storage_domains, storage_type, template, total_size, uses_scsi_reservation, vm, vms, wipe_after_delete)
  16167         self.content_type = content_type
  16168         self.disk_profile = disk_profile
> 16169         self.format = format
  16170         self.image_id = image_id
  16171         self.initial_size = initial_size

/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/ovirtsdk4/types.pyc in format(self, value)
  16263         Sets the value of the `format` property.
  16264         """
> 16265         Struct._check_type('format', value, DiskFormat)
  16266         self._format = value
  16267 

/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/ovirtsdk4/__init__.pyc in _check_type(attribute, value, expected)
    166                     attribute=attribute,
    167                     actual=actual.__name__,
--> 168                     expected=expected.__name__,
    169                 ))
    170 

TypeError: The type 'str' isn't valid for attribute 'format', it must be 'DiskFormat'
```